### DISCUSSION TITLE
Monaco editor lazy loading

### DISCUSSION THREAD

**Discussion by DaniShMyGit** [microsoft/monaco-editor#3647] [Q&A]
Hi all,
I'm not sure the title is correct but I'll try to explain my problem:
1. I have a parent component which retrieves data from a server.

` private showTestCases(): void {

    this.service.readTestCases()
      .pipe(
        tap((testCases) => {
          this.testCases = testCases.elements;
          if (this.testCases.length > 0) {
            this.activeId = this.testCases[0].testCaseRunResults.length > 0
              ? this.testCases[0].name
              : '';
            this.selectedTestCaseRunResult = this.testCases[0]?.testCaseRunResults[0];
          }
        }),
        takeUntil(this.destroy$),
      ).subscribe(() => {
        this.changeDetectorRef.detectChanges();
    });
  }`

2. `activeId` used to mark the first accordion panel as opened
3. I show the data on NgbAccordion:

` <ngb-accordion [activeIds]="activeId">

      <ngb-panel *ngFor="let testCase of testCases;let i = index;" [id]="testCase.name">
          <ng-template ngbPanelContent>
             <div  *ngIf="selectedTestCaseRunResult">
              <app-asset-test-data [selectedTestCaseRunResult]="selectedTestCaseRunResult"></app-asset-test-data>
            </div>
           </ng-template>
      </ngb-panel>
  </ngb-accordion>  `
  
  3. In the inner child component `app-asset-test-data` I'm using monaco editor to display the `selectedTestCaseRunResult` ( `selectedTestCaseRunResult` is the data we injecting to the child component):
  
  `  <ngx-monaco-editor class="code-editor"
  
    [options]="editorOptions" [ngModel]="(selectedTestCaseRunResult.input) | json">
  </ngx-monaco-editor>`
  
  4. I can see the injection passed successfully, I also can see in the inspect window that the monaco gets my input as a value but the monaco appeared as empty in the UI. It's only works if I'm adding a timer/delay before injecting the `selectedTestCaseRunResult`. Is it a monaco bug? Is there a way I can bypass this issue?

**Comment by Andrew3005** [comment]
I had the similar problem, after using another editor it disappear
